cmd/minikube/cmd/start.go

```diff
@@ -502,25 +505,33 @@ func selectImageRepository(mirrorCountry string, k8sVersion string) (bool, strin
 // validerUser validates minikube is run by the recommended user (privileged or regular)
 func validateUser() {
 	u, err := user.Current()
+	if err != nil {
+		glog.Errorf("Error getting the current user: %v", err)
+		return
+	}
+
 	d := viper.GetString(vmDriver)
-	// Check if minikube needs to run with sudo or not.
-	if err == nil {
-		if d == constants.DriverNone && u.Name != "root" {
-			exit.UsageT(`Please run with sudo. the vm-driver "{{.driver_name}}" requires sudo.`, out.V{"driver_name": constants.DriverNone})
-		} else if u.Name == "root" && !(d == constants.DriverHyperv || d == constants.DriverNone) {
-			out.T(out.WarningType, "Please don't run minikube as root or with 'sudo' privileges. It isn't necessary with {{.driver}} driver.", out.V{"driver": d})
-		}
+	useForce := viper.GetBool(force)
 
-	} else {
-		glog.Errorf("Error getting the current user: %v", err)
+	if d == constants.DriverNone && u.Uid != "0" && !useForce {
+		exit.WithCodeT(exit.Permissions, `The "{{.driver_name}}" driver requires root privileges. Please run minikube using 'sudo minikube --vm-driver={{.driver_name}}'.`, out.V{"driver_name": d})
 	}
 
+	if d == constants.DriverNone || u.Uid != "0" {
```

Comment:
Wouldn't this always exit, for driver none ? Or maybe that is the "happy path" of validateUser

Reply:
Never mind, the return will just go the same place as before (the checks were reversed)